### PR TITLE
Roomlog: Support getting total linecount

### PR DESCRIPTION
**Pull request by AnnikaCodes**
As discussed in PMs with Zarel

### CODE REVIEW

**Comment by Zarel** [comment]
This doesn't seem to include any changes to the repeats plugin. Do you want to add that here? Or are you intentionally splitting it up into separate PRs? (That's fine, too, I just wanted to make sure leaving it off here wasn't a mistake.)

File: server/roomlogs.ts [hunk 1]
```diff
@@ -265,9 +269,17 @@ export class Roomlog {
...
-			this.log.splice(0, this.log.length - 100);
+			const truncationLength = this.log.length - 100;
+			this.log.splice(0, truncationLength);
+			this.numTruncatedLines += truncationLength;
 		}
 	}
+	/**
+	 * Returns the total number of lines in the roomlog, including truncated lines.
+	 */
+	getLinecount() {
```
Comment (Zarel) [inline server/roomlogs.ts:280]: I'd prefer `getLineCount` with an uppercase `C`.

**Review by Zarel — APPROVED** [review]
Everything else looks good. I believe this crashes hotpatch, though?

**Comment by AnnikaCodes** [comment]
> This doesn't seem to include any changes to the repeats plugin. Do you want to add that here? Or are you intentionally splitting it up into separate PRs? (That's fine, too, I just wanted to make sure leaving it off here wasn't a mistake.)

The Repeats plugin will start working properly if the logic for nth message handlers in `rooms.ts` starts working.

And yeah, this can't be hotpatched. I'm happy to monkeypatch it, though, next time I hotpatch chat.

**Comment by Zarel** [comment]
Oh, huh, I didn't look closely enough to see how that worked.

Yeah, I approve everything here. Feel free to merge and monkeypatch this yourself.

**Pull request merged**